packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -4,6 +4,9 @@ export default function ({ types: t }) {
       if (t.isRestProperty(property)) {
         return true;
       }
+      if (t.isObjectProperty(property) && t.isObjectPattern(property.value)) {
```

Comment:
Nice, makes me wonder if there are other possible ways to have a restproperty - if so maybe we can consider doing a `path.traverse` and visiting RestProperty like in https://github.com/christophehurpeau/babel/blob/15fcc7c4e7765046aeb6146bc3501681cf309a78/packages/babel-plugin-transform-object-rest-spread/src/index.js#L91-L92